serverlessworkflow/modules/ROOT/pages/testing-and-troubleshooting/mocking-http-cloudevents-with-wiremock.adoc

```diff
@@ -1 +1,291 @@
+<7> Start the server before the tests executed.
+<8> Stub the mock API response. It accepts a MappingBuilder instance that we can use to build API mapping information such as URL, request parameters and body, headers, authorization etc.
+<9> `@AfterAll` annotation used to signal that the annotated method should be executed after all the tests execution.
+<10> Stop the server after all test execution.
+<11> {awaitility_class}[`await()`] added to waiting for asynchronous operations.
+<12> {wiremock_verifying}[`verify`] if the request has hit the mock API with the expected event content.
+
+[NOTE]
+====
+It is very important to start the server before the tests execute, and stop the server after the tests finish.
+We can reset the mock stubs in between the tests.
+====
+
+
+=== Configure the Quarkus test
+The test application needs to be configured to use the WireMockServer as the sink.
+The test {kogito_sw_order_processing_example_test_application_properties}[`application.properties`] file must contain the reference to the WireMockServer created.
+
+.Test {kogito_sw_order_processing_example_test_application_properties}[`application.properties`] sink connection property
+[source,properties]
+----
+mp.messaging.outgoing.kogito_outgoing_stream.url=http://0.0.0.0:8181 <1>
+----
+<1> The port needs to match with the one passed in the WireMockServer creation moment in the test class created.
+
+=== Execute Testing
+
+To run the tests execute the following command:
+--
+[source,shell]
+----
+mvn clean verify
+----
+--
+
+=== Testing execution cycle example
+
+The testing case followed in this document is based on the {kogito_sw_order_processing_example}[`serverless-workflow-order-processing`] example application.
+
+The {kogito_sw_order_processing_example}[example] is composed by three workflows:
+
+.Example Order Processing workflows
+image::testing-and-troubleshooting/order-example-worflows.png[]
+
+The main workflow process the incoming Order event and start a parallel state calling two subflows: Fraud Handling and Shipping Handling.
+The workflow will end once **both** subflows end.
+
+Fraud Handling will produce a new `FraudEvaluation` event if the order is above 1000 USD.
+Any other system or service in the architecture can then read this event and react upon it, like canceling the order for example.
+
+In parallel, regarding or not the order would need fraud evaluation, the workflow will produce events classifying the required Shipping service: International or Domestic.
+For this example, domestic shipping is any order with address within US.
+
+The event flow between components in the example application is shown in this diagram
+
+.Example event flow between components
+image::testing-and-troubleshooting/example-components-interaction.png[]
+
+Here, how the testing components replicates these relationships to verify the events received by the sink
+
+.Example testing events flow.
+image::testing-and-troubleshooting/testing-eventing-sink.png[]]
```

Comment:
I think in this image you can remove the "broker" since the events flow directly to the mock.